patroni/config.py

```diff
@@ -9,12 +9,31 @@
 from collections import defaultdict
 from copy import deepcopy
 from patroni.dcs import ClusterConfig
-from patroni.postgresql.config import ConfigHandler
+from patroni.postgresql.config import ConfigHandler, parse_dsn
 from patroni.utils import deep_compare, parse_bool, parse_int, patch_config
 from requests.structures import CaseInsensitiveDict
 
 logger = logging.getLogger(__name__)
 
+AUTH_ALLOWED_PARAMETERS = (
+    # universal
+    'username',
+    'password',
+    # postgresql only
+    'sslmode',
+    'sslcert',
+    'sslkey',
+    'sslrootcert',
+    'sslcrl',
+    # etcd only
```

Comment:
We don't need to put anything related to etcd or restapi here.
The problem is that ENV variable names are not always consistent with structures in YAML file:
```
PATRONI_RESTAPI_USERNAME=user
PATRONI_RESTAPI_PASSWORD=passwd
```
vs
```yaml
restapi:
  authentication:
    username: user
    password: passwd
```

It doesn't break the code only because other ENV variable names are already "consumed" by [_set_section_values()](https://github.com/zalando/patroni/pull/1134/files#diff-188997f9c97e1dac6c37391a509d13c9R252) a few lines earlier than [_get_auth()](https://github.com/zalando/patroni/pull/1134/files#diff-188997f9c97e1dac6c37391a509d13c9L261)

Reply:
Great, thanks for the explanation! I condensed the `tuple` to only the relevant PostgreSQL parameters.